cmd/pctl/main.go

```diff
@@ -114,3 +177,30 @@ func printProfile(profile profilesv1.ProfileDescription) error {
 	fmt.Println(string(out))
 	return nil
 }
+
+// createFilesForArtifacts will create files for all the artifacts that profiles generated.
+func createFilesForArtifacts(artifacts []runtime.Object) error {
+	e := kjson.NewSerializerWithOptions(kjson.DefaultMetaFactory, nil, nil, kjson.SerializerOptions{Yaml: true, Strict: true})
+	generateOutput := func(i int, o runtime.Object) error {
+		filename := fmt.Sprintf("%s_%d_resource.%s", o.GetObjectKind().GroupVersionKind().Kind, i, "yaml")
+		f, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+		if err != nil {
+			return err
+		}
+		defer func(f *os.File) {
+			if err := f.Close(); err != nil {
+				fmt.Printf("Failed to properly close file %s\n", f.Name())
+			}
+		}(f)
+		if err := e.Encode(o, f); err != nil {
+			return err
+		}
+		return nil
+	}
+	for i, a := range artifacts {
+		if err := generateOutput(i, a); err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
does this logic belong in `main`? Makes it harder to unit test

Reply:
Yeah, no. I need to extract this and write some nice tests for it somewhere. :) You're right.